samples/dnn/CMakeLists.txt

```diff
@@ -14,18 +14,22 @@ if(NOT BUILD_EXAMPLES OR NOT OCV_DEPENDENCIES_FOUND)
 endif()
 
 function(download_net name commit hash)
-  set(DNN_FACE_DETECTOR_MODEL_DOWNLOAD_DIR "${CMAKE_CURRENT_LIST_DIR}/face_detector")
   if(COMMAND ocv_download)
     ocv_download(FILENAME ${name}
                HASH ${hash}
                URL
                  "$ENV{OPENCV_DNN_MODELS_URL}"
                  "${OPENCV_DNN_MODELS_URL}"
                  "https://raw.githubusercontent.com/opencv/opencv_3rdparty/${commit}/"
-               DESTINATION_DIR ${DNN_FACE_DETECTOR_MODEL_DOWNLOAD_DIR}
+               DESTINATION_DIR "${OpenCV_BINARY_DIR}/face_detector"
```

Comment:
It is located in the source directory for a reason.
There are other files which required to run sample.
Python sample goes broken (they are run from source directory).